sdk/compiler/daml-extension/src/language_client.ts

```diff
@@ -355,6 +355,26 @@ export class DamlLanguageClient {
     throw new Error("Failed to locate assistant.");
   }
 
+  // Once an assistant has been selected, we can find out the sdk version
+  // however if the sdk version is too low, it overrides the selected assistant
+  // this wrapper function handles that edge case
+  static async findAssistantCommandAndSdkVersion(
+    rootPath: string,
+    config: vscode.WorkspaceConfiguration,
+  ): Promise<[string, string | undefined]> {
+    const [command, isUsingDPM] =
+      DamlLanguageClient.findAssistantCommand(config);
+    const sdkVersion = isUsingDPM
+      ? await DamlLanguageClient.getSdkVersionDPM(command, rootPath)
+      : await DamlLanguageClient.getSdkVersionDamlAssistant(command, rootPath);
+
+    if (sdkVersion && sdkVersion != "0.0.0" && semver.lt(sdkVersion, "3.4.0")) {
+      const damlPath = DamlLanguageClient.findDamlCommand();
+      if (!damlPath) throw new Error("Failed to locate assistant.");
```

Comment:
I think this error could use a bit of clearing up:
```suggestion
      if (!damlPath) throw new Error("Failed to locate Daml assistant. A Daml assistant must be installed in order to use SDK versions <= 3.3.");
```